File: basic/src/main/java/com/yuanshenbin/basic/call/Callback.java
package com.yuanshenbin.basic.call;

/**
 * author : yuanshenbin
 * time   : 2018/4/5
 * desc   :
 */

public abstract class Callback<T> {

    public abstract void ok(T t);

    public void cancel(T t) {

    }

    public void close(){

    }
}

File: basic-kotlin/src/main/java/com/yuanshenbin/basic/call/Callback.kt
package com.yuanshenbin.basic.call

/**
 * author : yuanshenbin
 * time   : 2018/4/5
 * desc   :
 */
abstract class Callback<T> {
    abstract fun ok(t: T)
    open fun cancel(t: T) {}
    open fun close() {}
}
